python/tk_multi_reviewsubmission/renderer.py

```diff
@@ -1,187 +1,210 @@
-            output_node = self.__create_output_node(output_path)
-            output_node.setInput(0, scale)
-        finally:
-            group.end()
-    
-        if output_node:
-            # Make sure the output folder exists
-            output_folder = os.path.dirname(output_path)
-            self.__app.ensure_folder_exists(output_folder)
-            
-            # Render the outputs, first view only
-            nuke.executeMultiple([output_node], ([first_frame-1, last_frame, 1],), [nuke.views()[0]])
-
-        # Cleanup after ourselves
-        nuke.delete(group)
-        
-        
-    def __create_scale_node(self, width, height):
-        """
-        Create the Nuke scale node to resize the content.
-        """
-        scale = nuke.nodes.Reformat()
-        scale["type"].setValue("to box")
-        scale["box_width"].setValue(width)
-        scale["box_height"].setValue(height)
-        scale["resize"].setValue("fit")
-        scale["box_fixed"].setValue(True)
-        scale["center"].setValue(True)
-        scale["black_outside"].setValue(True)
-        return scale
-
-    def __create_output_node(self, path):
-        """
-        Create the Nuke output node for the movie.
-        """
         # get the Write node settings we'll use for generating the Quicktime
-        wn_settings = self.__app.execute_hook_method("codec_settings_hook", 
-                                                     "get_quicktime_settings")
-
-        node = nuke.nodes.Write(file_type=wn_settings.get("file_type"))
-        
-        # apply any additional knob settings provided by the hook. Now that the knob has been 
-        # created, we can be sure specific file_type settings will be valid.
-        for knob_name, knob_value in wn_settings.iteritems():
-            if knob_name != "file_type":
-                node.knob(knob_name).setValue(knob_value)
-        
-        # Don't fail if we're in proxy mode. The default Nuke publish will fail if
-        # you try and publish while in proxy mode. But in earlier versions of 
-        # tk-multi-publish (< v0.6.9) if there is no proxy template set, it falls 
-        # back on the full-res version and will succeed. This handles that case
-        # and any custom cases where you may want to send your proxy render to 
-        # screening room. 
-        root_node = nuke.root()
-        is_proxy = root_node['proxy'].value()
-        if is_proxy:
-            self.__app.log_info("Proxy mode is ON. Rendering proxy.")
-            node["proxy"].setValue(path.replace(os.sep, "/"))
+        writenode_quicktime_settings = self.__app.execute_hook_method("codec_settings_hook",
+                                                                      "get_quicktime_settings")
+
+        render_script_path = os.path.join(self.__app.disk_location, "hooks",
+                                          "nuke_batch_render_movie.py")
+        ctx = self.__app.context
+
+        shotgun_context = {}
```

Comment:
Please use `sgtk.context.serialize()` and `sgtk.context.deserialize()` and call this variable `serialized_context` for consistency with other tools.

Reply:
Done. Also serialising all other variables passed to the subprocess using pickle.